### DISCUSSION TITLE
How to pass parameters to entrypoint like the functionality provided by docker exec form

### DISCUSSION THREAD

**Discussion by Bidaya0** [batect/batect#1199] [Q&A] [answered]
The entrypoint of dockerfile has the following two types
exec form
```
ENTRYPOINT ["executable", "param1", "param2"]
```
shell form
```
ENTRYPOINT command param1 param2
```

In most cases, the two are the same, but the exec form can append the command line parameters when docker run with the exec form.

This feature is useful when I am using batect in a django project, this is useful for using `` python manage.py`` to run command. 
Without that I can only warpper a set of tasks for the commands provided by python manage.py.
Is there a way out or better practice?

**Comment by charleskorn** [comment, marked as answer]
I'm not sure I 100% understand your question @Bidaya0 - if you could add an example of what you're trying to achieve, that might help me understand.

Here's some information that might help:

If you're specifying the entrypoint in your Dockerfile, whatever form you use there (exec or shell) will be used by Batect. If you're using the [`entrypoint` configuration option](https://batect.dev/docs/reference/config/containers/#entrypoint), this will be converted to the exec form by Batect for you.

The command for your container (whether from the `CMD` instruction in your image or the `command` configuration option for your container or task from your Batect configuration) will be appended to the entrypoint to form the final command line executed by Docker. 

You can also pass further arguments on the command line, for example if `command` is `python manage.py` and you run `./batect build -- --option=value`, the final command line would be `python manage.py --option=value`.

**Reply by Bidaya0** [reply]
> I'm not sure I 100% understand your question @Bidaya0 - if you could add an example of what you're trying to achieve, that might help me understand.
> 
> Here's some information that might help:
> 
> If you're specifying the entrypoint in your Dockerfile, whatever form you use there (exec or shell) will be used by Batect. If you're using the [`entrypoint` configuration option](https://batect.dev/docs/reference/config/containers/#entrypoint), this will be converted to the exec form by Batect for you.
> 
> The command for your container (whether from the `CMD` instruction in your image or the `command` configuration option for your container or task from your Batect configuration) will be appended to the entrypoint to form the final command line executed by Docker.
> 
> You can also pass further arguments on the command line, for example if `command` is `python manage.py` and you run `./batect build -- --option=value`, the final command line would be `python manage.py --option=value`.

Thanks, I wrongly used `command` and `entrypoint` command.
when I use `entrypoint`, it show
```
batect.config.ContainerCommandResolutionException: Additional command line arguments for the task have been provided, but neither the task (manage) nor the main task container (dongtai-webapi) have an explicit command in the configuration file.
``` 
but after I changed  it to `command`,
it works fine .
